doc/tor.md

```diff
@@ -30,7 +45,11 @@ outgoing connections, but more is possible.
 	-onlynet=onion  Make outgoing connections only to .onion addresses. Incoming
 	                connections are not affected by this option. This option can be
 	                specified multiple times to allow multiple network types, e.g.
-	                ipv4, ipv6, or onion.
+	                ipv4, ipv6 or onion. If you use this option with values other
+	                than onion you *cannot* disable onion connections; outgoing onion
+	                connections will be enabled when you use -proxy or -onion. Use
+	                -noonion or -onion=0 if you want to be sure there are no outbound
+	                onion connections over the default proxy or your defined -proxy.
```

Comment:
Huh! :-( This is indeed a strange behavior, but I think something like the following belongs to `bitcoind -help` rather than `doc/tor.md`:

> If -onlynet= is used with other networks and -onlynet=onion is not among them, but -onion= or -proxy is provided then outbound onion connections will still be made! Use -noonion or -onion=0 to disable outbound onion connections in this case.

Given that this is a guide on how to enable and configure Tor, not how to disable it, I would suggest to drop the last part and leave it to just:

```
-onlynet=onion  Make outgoing connections only to .onion addresses. Incoming
	        connections are not affected by this option.
```

Reply:
That would be removing needed information that was present before this PR!

Reply:
Dropped the commit completely.

Reply:
@vasild 
> -onlynet=onion  Make outgoing connections only to .onion addresses. Incoming
        connections are not affected by this option.

I guess its hard to grasp and i was :woman_facepalming:  too when i stumbled about, but what we believed the code does from src comments or doc and almost any post i am aware about bitcoin and Tor and social media is
not what really happens when assuming option do what they say and u follow the docs.

only -noonion or -onion=0 can prevent that, onlynet= is buggy since  #7553 (2016)
Similar but also fact is that although seerders are not called directly in IP if the DNS returns are random that is not the case when bootstrap over Tor here the seeders have full control and can be sure ( or ther MITM'S or proxys)  that the first entry in the DNS return a exit gets will be called since socks5 works that way all seeders will be called in a distinct sequence over exist and then ADDR gossip that is highly disturbing give the timing correlation bogus Tor relays and exits can do.
Tacking in account that all that happens and wrong configured at bootstrap at least once, *pure* Tor nodes are quite f***ed if they not aware.

Reply:
Brought back the commits and added also to the -onlynet help in #20757, please add suggestions or ACKs there.